File: src/test/java/com/planit/pages/HomePage.java
package com.planit.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class HomePage extends BasePage {

    private static final By NAV_CONTACT = By.linkText("Contact");
    private static final By NAV_SHOP    = By.linkText("Shop");

    public HomePage(WebDriver driver) {
        super(driver);
    }

    public ContactPage goToContact() {
        click(NAV_CONTACT);
        return new ContactPage(driver);
    }

    public ShopPage goToShop() {
        click(NAV_SHOP);
        return new ShopPage(driver);
    }
}

File: PlanitAutomation/Pages/HomePage.cs
using Microsoft.Playwright;

namespace PlanitAutomation.Pages;

/// <summary>
/// Represents the Jupiter Toys home page.
/// Provides navigation helpers to other pages.
/// </summary>
public class HomePage(IPage page) : BasePage(page)
{
    private const string NavContact = "a:text('Contact')";
    private const string NavShop    = "a:text('Shop')";

    public async Task<ContactPage> GoToContactAsync()
    {
        await ClickAsync(NavContact);
        return new ContactPage(Page);
    }

    public async Task<ShopPage> GoToShopAsync()
    {
        await ClickAsync(NavShop);
        return new ShopPage(Page);
    }
}
